kasa/smartdimmer.py

```diff
@@ -41,19 +43,42 @@ def brightness(self) -> int:
         return int(sys_info["brightness"])
 
     @requires_update
-    async def set_brightness(self, value: int):
-        """Set the new dimmer brightness level in percentage."""
+    async def set_brightness(self, brightness: int, *, transition: int = None):
+        """Set the new dimmer brightness level in percentage.
+
+        :param int transition: transition duration in milliseconds.
+            Will turn on the light if brightness > 0,
```

Comment:
I think this should explicitly state that these will happen only if `transition` is used.